pdns/pdns_recursor.cc

```diff
@@ -1720,6 +1712,14 @@ void indirectDoProcessUDPQuestion(void *p)
 
 string *doProcessUDPQuestion(const std::string& question, const ComboAddress& fromaddr, const ComboAddress& destaddr, struct timeval tv, int fd)
 {
+  if(MT->numProcesses() > g_maxMThreads) {
+    if(!g_quiet)
+      L<<Logger::Notice<<t_id<<" ["<<MT->getTid()<<"/"<<MT->numProcesses()<<"] DROPPED question from "<<fromaddr.toStringWithPort()<<", over capacity"<<endl;
+
+    g_stats.overCapacityDrops++;
+    return 0;
+  }
```

Comment:
ah, yes. No packetcache on TCP, that's why that code was simpler. I worried about this, thanks. Will indeed have to 'refactor' as I initally wrote down.